lib/system.nim

```diff
@@ -4239,3 +4239,8 @@ when defined(genode):
       componentConstructHook(env)
         # Perform application initialization
         # and return to thread entrypoint.
+
+const systemWasImported* = true
+  ## Use case: `when declared(systemWasImported): export foo` where foo is in a
+  ## `bar.nim`, where system.nim has "include bar" and another module has
+  ## `import bar` ; see example in system/helpers.nim
```

Comment:
What about:
`system/inclhelpers.nim`, which doesn't export anything and gets included by system
`syste/helpers.nim`, which includes `system/inclhelpers.nim` and exports the procs

We shouldn't use a hack that requires exporting symbols from `system.nim`.